src/sources/kubernetes_logs/parser/cri.rs

```diff
@@ -66,6 +67,9 @@ impl FunctionTransform for Cri {
 }
 
 fn normalize_event(log: &mut LogEvent) -> Result<(), NormalizationError> {
+    // Remove possible new_line tag
+    // for additional details, see https://github.com/timberio/vector/issues/8606
+    let _ = log.remove(NEW_LINE_TAG);
```

Comment:
Do we want to drop the last byte? I thought, in your case, the last byte could be `0a` (newline) but also be part of an unfinished unicode character.

Reply:
Hi @jszwedko! Sorry for the late answer
In our case, this last byte is the part of the unfinished unicode character, not a newline symbol.
I think, if the log message contains an `\n` symbol, e.g. newline, it will be escaped by the log driver.
In [sources](https://github.com/containers/conmon/blob/main/src/ctr_logging.c#L355) for CRI-O log driver, this additional `\n` byte inserted for partial line, therefore we want to drop it from the log message

Reply:
@ArtemTrofimushkin Ah I see, so the trailing `\n` will be guaranteed to be there for partial lines. Should we only trim it if the `multiline_tag` capture group is `P`? I'm wondering if `0a` (`\n`) would ever be a byte of a UTF-8 character? In that case `.` wouldn't actually match it.

Reply:
Because of this part `(?P<new_line_tag>\n?)` of regex, new_line_tag will always exists in LogEvent (in case of full message, value will be `false`). That's why it always droped from LogEvent. 
[It](https://datatracker.ietf.org/doc/html/rfc3629#section-3) seems, that unicode characters will not contain a newline byte

Reply:
Ah yes, that makes sense. Thanks for clarifying.